.github/workflows/docker.yml

```diff
@@ -1,136 +1,616 @@
-name: Docker
-
-# initially we would us on: [release] as well, the problem is that
-# the code in clowder would not know what branch the code is in,
-# and would not set the right version flags.
-
-# This will run when:
-# - when new code is pushed to master/develop to push the tags
-#   latest and develop
-# - when a pull request is created and updated  to make sure the
-#   Dockerfile is still valid.
-# To be able to push to dockerhub, this expects the following
-# secrets to be set in the project:
-# - DOCKERHUB_USERNAME : username that can push to the org
-# - DOCKERHUB_PASSWORD : password asscoaited with the username
+name: Docker GHA
+
 on:
-  push:
-    branches:
-      - master
-      - develop
+  # push:
+  #   branches:
+  #     - main
+  #   tags:
+  #     - "v*.*.*"
   pull_request:
-  merge_group:
-  issue_comment:
-    types:
-      - created
   workflow_dispatch:
+    inputs:
+      depends:
+        description: 'Create depends image'
+        required: true
+        type: boolean
+        default: true
+      r_version:
+        description: 'R version to use'
+        required: true
+        type: choice
+        default: "4.1"
+        options:
+          - 4.1
+          - 4.2
+          - 4.3
 
-# Certain actions will only run when this is the master repo.
+# set up the environment, either input or default
 env:
-  MASTER_REPO: PecanProject/pecan
-  DOCKERHUB_ORG: pecan
-  GITHUB_PAT: ${{ secrets.GITHUB_TOKEN }}
+  R_VERSION: ${{ github.event.inputs.r_version || '4.1' }}
```

Comment:
Is the `|| '4.1'` needed here? I was assuming the `default: "4.1"` on line 21 would ensure a value, but haven't tested my assumption.

Reply:
the 4.1 is needed because the inputs are ONLY set when using a dispatch. :(